src/analyticsdf/analyticsdataframe.py

```diff
@@ -198,6 +198,7 @@ def _is_len_match_list(list1, list1_name, list2, list2_name):
         col_name = self.predictor_matrix.columns.values.tolist()
         if _is_subset_list(predictor_name_list, "predictor_name_list", col_name) and \
             _is_len_match_list(predictor_name_list, "predictor", beta[1:], "beta"):
+            np.random.seed(0)   # set the random seed to make sure the error in pytest function is the same
```

Comment:
I don't think it's the right thing to do. Then there would be no randomness for using this function, all generated datapoints would be the same

Reply:
Thanks for pointing it out! I understand what you mean.
However, as long as the predictors are not exactly the same, which shouldn't be, the linear response vector is still random due to the inherent randomness of the predictors.
That being said, I'll try my best to see if I can design a reasonable pytest function without setting the random seed first. If I can't find out a solution, I'll create an issue that's open for discussion.